File: android/app/src/main/java/tech/orizon/ampara/audio/phrase/PipelineState.java
package tech.orizon.ampara.audio.phrase;

/**
 * Estados do pipeline de detecção (SPEC seção 7).
 * 
 * Descreve cada estágio com entrada, saída, timeout e comportamento.
 */
public enum PipelineState {

    /** Monitoramento idle — L1 roda contínuo */
    IDLE_MONITORING,

    /** L1 detectou evento acústico suspeito — preparando escalação */
    ACOUSTIC_SUSPICION,

    /** Buffer de confirmação está sendo montado (pre + post trigger) */
    CONFIRMATION_WINDOW,

    /** L3 analisando frases na janela de confirmação */
    PHRASE_ANALYSIS,

    /** L4 calculando riskScore e determinando ação */
    RISK_SCORING,

    /** Ação operacional sendo executada (gravação, pânico, etc.) */
    ACTION_TRIGGERED,

    /** Período de cooldown após ação executada */
    COOLDOWN,

    /** Modo B — escuta contínua low-power de comandos operacionais */
    COMMAND_LISTENING
}

File: ios/App/Plugins/Phrase/PipelineState.swift
import Foundation

/// Estados do pipeline de detecção — paridade com PipelineState.java
enum PipelineState: String {
    /// Monitoramento idle — apenas Layer 1 (acústica leve)
    case idleMonitoring = "IDLE_MONITORING"
    /// Suspeita acústica detectada — preparando buffer
    case acousticSuspicion = "ACOUSTIC_SUSPICION"
    /// Janela de confirmação aberta — coletando áudio
    case confirmationWindow = "CONFIRMATION_WINDOW"
    /// Análise de frase em andamento — Layer 3 (DTW)
    case phraseAnalysis = "PHRASE_ANALYSIS"
    /// Cálculo de risk score — Layer 4
    case riskScoring = "RISK_SCORING"
    /// Ação disparada — aguardando resolução
    case actionTriggered = "ACTION_TRIGGERED"
    /// Cooldown após ação
    case cooldown = "COOLDOWN"
    /// Escuta contínua de comandos (Mode B)
    case commandListening = "COMMAND_LISTENING"
}
